src/main/java/org/radarcns/management/web/rest/OAuthClientsResource.java

```diff
@@ -80,20 +93,110 @@
     @Autowired
     private HttpServletRequest servletRequest;
 
+    private static final String ENTITY_NAME = "oauthClient";
+    private static final String PROTECTED_KEY = "protected";
+
+    /**
+     * GET /api/oauthclients
+     *
+     * Retrieve a list of currently registered OAuth clients.
+     *
+     * @return the list of registered clients as a list of {@link ClientDetailsDTO}
+     */
     @GetMapping("/oauthclients")
     @Timed
     public ResponseEntity<List<ClientDetailsDTO>> getOAuthClients() {
         checkPermission(getJWT(servletRequest), OAUTHCLIENTS_READ);
-        return ResponseEntity.ok().body(clientDetailsMapper.clientDetailsToClientDetailsDTO(
-            clientDetailsService.listClientDetails()));
+        return ResponseEntity.ok().body(clientDetailsMapper
+                .clientDetailsToClientDetailsDTO(clientDetailsService.listClientDetails()));
     }
 
-    @GetMapping("/oauthclients/:id")
+    /**
+     * GET /api/oauthclients/:id
+     *
+     * Get details on a specific client.
+     *
+     * @param id the client id for which to fetch the details
+     * @return the client as a {@link ClientDetailsDTO}
+     */
+    @GetMapping("/oauthclients/{id}")
     @Timed
     public ResponseEntity<ClientDetailsDTO> getOAuthClientById(@PathVariable("id") String id) {
         checkPermission(getJWT(servletRequest), OAUTHCLIENTS_READ);
-        return ResponseEntity.ok().body(
-            clientDetailsMapper.clientDetailsToClientDetailsDTO(getOAuthClient(id)));
+        return ResponseEntity.ok().body(clientDetailsMapper
```

Comment:
Validate whether id is empty

Reply:
When `id` is left empty, the request will be routed to `/api/oauth-clients` and return all clients. An unknown client will throw a `CustomNotFoundException` and thus use the `ExceptionTranslator` to build a 404 response.

Reply:
This it should be `@RequestParam`. I don't think a path parameter can be empty. If it can be empty there is no point of having it as path parameter. There is also another api to do that. Following the practice in all the entities i would say it makes sense to have the validation.

Reply:
Sorry for the confusion. It is not necessary . I got confused with the below validation. For `PUT` we need additional validation before we proceed.

Reply:
Indeed, will add it!